### PR TITLE
Check Access Key Presence to avoid Build Cache errors on PR builds where Key is not present

### PR DESCRIPTION
**Pull request by runningcode**

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -26,7 +26,9 @@ buildCache {
 
     remote(gradleEnterprise.buildCache) {
         enabled = true
-        push = isCI
+        // Check access key presence to avoid build cache errors on PR builds when access key is not present
+        def accessKey = System.getenv("GRADLE_ENTERPRISE_ACCESS_KEY")
+        push = isCI && accessKey != null && !accessKey.isEmpty()
```
Comment (etiennestuder) [inline settings.gradle:31]: This can be simplified to isCi && accessKey.

Comment (etiennestuder) [inline settings.gradle:31]: If we do this for this repo, we should do it for the other repos, too. Of, if we only need/want one example, we should do it in the config samples repo (for groovy and Kotlin).

Comment (runningcode) [reply settings.gradle:31]: Agreed, we should do this for all the repos.

Comment (runningcode) [reply settings.gradle]: updated

Comment (etiennestuder) [reply settings.gradle]: But please be sure to verify it and not take my word for it.

Comment (runningcode) [reply settings.gradle]: Yup, this PR is from a fork, and there are no 403 errors or other remote build cache errors in the logs: https://github.com/gradle/common-custom-user-data-gradle-plugin/actions/runs/7003812403/job/19050268047

**Pull request merged**